Source/WebCore/svg/SVGSVGElement.cpp

```diff
@@ -668,8 +669,21 @@ float SVGSVGElement::intrinsicHeight() const
 
 AffineTransform SVGSVGElement::viewBoxToViewTransform(float viewWidth, float viewHeight) const
 {
-    if (!m_useCurrentView || !m_viewSpec)
-        return SVGFitToViewBox::viewBoxToViewTransform(currentViewBoxRect(), preserveAspectRatio(), viewWidth, viewHeight);
+    if (!m_useCurrentView || !m_viewSpec) {
+        auto currentViewBox = currentViewBoxRect();
+        auto pAR = preserveAspectRatio();
+
+        // If we synthesized a viewBox (no explicit viewBox but embedded through SVGImage),
+        // we should also synthesize preserveAspectRatio="none" to allow stretching.
+        if (viewBox().isEmpty() && !currentViewBox.isEmpty() && isEmbeddedThroughSVGImage(*this)) {
+            // Create a synthetic preserveAspectRatio with align="none".
+            SVGPreserveAspectRatioValue syntheticPAR;
+            syntheticPAR.setAlign(SVGPreserveAspectRatioValue::SVG_PRESERVEASPECTRATIO_NONE);
+            return SVGFitToViewBox::viewBoxToViewTransform(currentViewBox, syntheticPAR, viewWidth, viewHeight);
+        }
+
+        return SVGFitToViewBox::viewBoxToViewTransform(currentViewBox, pAR, viewWidth, viewHeight);
```

Comment:
Please do not use abbreviations like pAR.
```
auto preserveAspectRatio = this->preserveAspectRatio();
```

Reply:
I think this can be written like this:
```
    if (viewBox().isEmpty() && !currentViewBox.isEmpty() && isEmbeddedThroughSVGImage(*this)) {
        auto preserveAspectRatio = SVGPreserveAspectRatioValue(SVGPreserveAspectRatioValue::SVG_PRESERVEASPECTRATIO_NONE, SVGPreserveAspectRatioValue::SVG_MEETORSLICE_MEET);
        return SVGFitToViewBox::viewBoxToViewTransform(currentViewBox, preserveAspectRatio, viewWidth, viewHeight);
        }

    return SVGFitToViewBox::viewBoxToViewTransform(currentViewBox, preserveAspectRatio(), viewWidth, viewHeight);
```

Reply:
@shallawa - Done.